audioswitch/src/main/java/com/twilio/audioswitch/bluetooth/BluetoothHeadsetManager.kt

```diff
@@ -170,7 +170,10 @@ internal constructor(
         if (hasPermissions()) {
             headsetListener = null
             bluetoothAdapter.closeProfileProxy(BluetoothProfile.HEADSET, headsetProxy)
-            context.unregisterReceiver(this)
+            // because start may have been called before bluetooth permissions were enabled
+            try {
+                context.unregisterReceiver(this)
+            } catch (_: IllegalArgumentException) { /* silently fall though */ }
```

Comment:
I'm OK with this implementation, but I'd prefer if we instead use a member variable to track if we registered, and check that before calling `unregisterReceiver()`.

Reply:
ok, updated